src/output/scanline_output_file.rs

```diff
@@ -134,10 +159,73 @@ impl<'a> ScanlineOutputFile<'a> {
             Ok(())
         }
     }
 
+    /// Writes image data from the given FrameBuffer.
+    ///
+    /// The passed FrameBuffer may have a different vertical resolution than the
+    /// image, but must have the same horizontal resolution.  Multiple calls to
+    /// this method in sequence will incrementally write subsequent vertical
+    /// chunks of the image.
+    ///
+    /// If the FrameBuffer has fewer scanlines than the remaining scanlines in
+    /// the image, then only that many scanlines will be written.  If the
+    /// FrameBuffer has more scanlines than remain in the image, then the
+    /// only the remaining number of scanlines will be written from the
+    /// FrameBuffer, and the image will be complete.
+    ///
+    /// On success returns the number of scanlines written.
+    pub fn write_pixels_incremental(&mut self, framebuffer: &FrameBuffer) -> Result<(u32)> {
+        // Validation
+        if self.scanlines_written == self.header().data_dimensions().1 {
+            return Err(Error::Generic("All scanlines have already been \
+                written, cannot do another incremental write"
+                                              .to_string()));
+        }
+
+        if self.header().data_dimensions().0 != framebuffer.dimensions().0 {
+            return Err(Error::Generic(format!("framebuffer width {} does not match\
+                                              image width {}",
+                                              framebuffer.dimensions().0,
+                                              self.header().data_dimensions().0)));
+        }
+
+        framebuffer.validate_channels_for_output(self.header())?;
+
+        // Set up the framebuffer with the image
+        let scanline_write_count = min(self.header().data_dimensions().1 - self.scanlines_written,
+                                       framebuffer.dimensions().1);
+
+        let mut error_out = ptr::null();
+
+        let error = unsafe {
+            let offset_fb = CEXR_FrameBuffer_copy_and_offset_scanlines(framebuffer.handle(),
+                                                                       self.scanlines_written);
+            let err = CEXR_OutputFile_set_framebuffer(self.handle, offset_fb, &mut error_out);
+            CEXR_FrameBuffer_delete(offset_fb);
+            err
+        };
+        if error != 0 {
+            let msg = unsafe { CStr::from_ptr(error_out) };
+            return Err(Error::Generic(msg.to_string_lossy().into_owned()));
+        }
+
+        // Write out the image data
+        let error = unsafe {
+            CEXR_OutputFile_write_pixels(self.handle, scanline_write_count as i32, &mut error_out)
+        };
+        if error != 0 {
+            let msg = unsafe { CStr::from_ptr(error_out) };
+            Err(Error::Generic(msg.to_string_lossy().into_owned()))
+        } else {
+            self.scanlines_written += scanline_write_count;
```

Comment:
Yeah, checked the C++ API.  In the documentation for the `writePixels` method, it says:

> To produce a complete and correct file, exactly m scan lines must be written, where m is equal to header().dataWindow().max.y - header().dataWindow().min.y + 1.